tasks/prime_workflow.py

```diff
@@ -0,0 +1,204 @@
+            )
+
+    def _get_array_name(self, object_key):
+        # Based on the object key, find the right array for nested objects in the mto
+        object_array_map = {
+            MTO_SERVICE_ITEM: "mtoServiceItems",
+            MTO_SHIPMENT: "mtoShipments",
+            PAYMENT_REQUEST: "paymentRequests",
+        }
+        try:
+            array_key = object_array_map[object_key]
+        except KeyError:
+            logger.exception(f"An unexpected type of {object_key} was passed into _add_to_mto_nested")
+            return None
+        return array_key
+
+    def _update_mto_nested(self, object_key, old_data, new_data):
+        """Replaces an object in an array directly nested under the MTO.
+        Can be used for mtoServiceItems, mtoShipments and mtoPaymentRequests
+        as they are all held in top level arrays.
+        Cannot be used for mtoAgent as it is nested further within the mtoShipment
+        """
+        # Replacement object should be associated with the current move, we assume it has a moveTaskOrderID field to check
+        if old_data["moveTaskOrderID"] != self.current_move["id"]:
+            raise ImplementationError(
+                (
+                    f"Cannot add object to store because its moveTaskOrderID of {old_data['moveTaskOrderID']} "
+                    f"does not match the current move's id of {self.current_move['id']}."
+                )
+            )
+
+        # Get the nested array from the current move, or None if it doesn't exist
+        array_key = self._get_array_name(object_key)
+        nested_array = self.current_move.get(array_key, None)
+
+        # If nested array does not exist or is empty, raise error, we should have found the object to replace
+        if nested_array is None or nested_array == []:
+            raise ImplementationError(f"Cannot find {object_key} of id {old_data['id']} to replace in {array_key}")
+
+        # Find item and replace it!
+        found = False
+        for idx, elem in enumerate(nested_array):
+            if elem["id"] == old_data["id"]:
+                nested_array[idx] = new_data
+                found = True
+                break
+
+        if not found:
+            raise ImplementationError(f"Cannot find {object_key} of id {old_data['id']} to replace in {array_key}")
+
+        self.current_move[array_key] = nested_array
+
+    def _get_mto_nested(self, object_key, id):
+        """Return object of type requested. If no id is provided return a random element.
+        If id is provided, return the specific element or raise error
+        """
+
+        # Get the nested array from the current move, or None if it doesn't exist
+        array_key = self._get_array_name(object_key)
+        nested_array = self.current_move.get(array_key, None)
+
+        # If nested array does not exist or is empty, return None
+        if nested_array is None or nested_array == []:
```

Comment:
Same as above, can be: `if not nested_array:`

Reply:
Done

Reply:
I mentioned this above, but `if not nested_array` will catch an empty list, so you don't need to also check for an empty list.

Reply:
Clearly my reading capacity is terrible on "monday" mornings.